testtools/_deferredmatchers.py

```diff
@@ -0,0 +1,184 @@
+# Copyright (c) testtools developers. See LICENSE for details.
+
+"""Matchers that operate on synchronous Deferreds.
+
+A "synchronous" Deferred is one that does not need the reactor or any other
+asynchronous process in order to fire.
+
+Normal application code can't know when a Deferred is going to fire, because
+that is generally left up to the reactor. Unit tests can (and should!) provide
+fake reactors, or don't use the reactor at all, so that Deferreds fire
+synchronously.
+
+These matchers allow you to make assertions about when and how Deferreds fire,
+and about what values they fire with.
+
+"""
+
+# TODO: None of these are published yet. Decide where & how to make them
+# public.
+from testtools.compat import _u
+from testtools.matchers import Mismatch
+
+from ._deferred import failure_content, on_deferred_result
+
+
+class _NoResult(object):
+    """Matches a Deferred that has not yet fired."""
+
+    @staticmethod
+    def _got_result(deferred, result):
+        return Mismatch(
+            _u('No result expected on %r, found %r instead'
+               % (deferred, result)))
+
+    def match(self, deferred):
+        """Match ``deferred`` if it hasn't fired."""
+        return on_deferred_result(
+            deferred,
+            on_success=self._got_result,
+            on_failure=self._got_result,
+            on_no_result=lambda _: None,
+        )
+
+
+NO_RESULT = _NoResult()
+"""Match a Deferred that has not yet fired.
+
+For example, this will pass::
+
+    assert_that(defer.Deferred(), NO_RESULT)
```

Comment:
Okay I guess the property of the model that suggests the naming divergence between `NO_RESULT` and the others is that `NO_RESULT` is a singleton matcher, not a function of one argument that returns a matcher.